torch/_meta_registrations.py

```diff
@@ -2105,7 +2105,10 @@ def _cudnn_rnn(
     output = input.new_empty(out_shape)
 
     cell_shape = [num_layers * num_directions, mini_batch, hidden_size]
-    cy = cx.new_empty(0 if cx is None else cell_shape)
+    if cx is None:
+        cy = torch.empty(0, device=input.device)
```

Comment:
Can we double check this is indeed what the kernel does internally (and not something like input.new_empty)

Reply:
Yea, either way works well in this case, as we need an empty tensor of size 0 on the specific device.